src/components/Form/index.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getResults } from '../../actions';
+import { Redirect } from 'react-router-dom';
+import './style.css';
+
+function Form() {
+
+    const [username, setUsername] = useState('');
+    const [redirect, setRedirect] = useState()
+    const dispatch = useDispatch();
+
+    function updateInput(e) {
+        setUsername(e.target.value);
+    }
+
+   async function handleSubmit(e) {
+        try {
+            e.preventDefault();
+            await getResults(dispatch, username);
+            setRedirect(<Redirect to='/user' />)
+        }catch(err){
+            console.log(err.message)
+        }
+
+        {/* { error!==false || ? <p role='alert'>{ error }</p> : <Redirect to='/user' /> } */ }
+        setUsername('');
+    }
+
+    return (
+        <>
+        <form aria-label="username-form" onSubmit={handleSubmit}>
+            <input aria-label="username" type="text" placeholder='username' value={username} onChange={updateInput} />
+            <input type="submit" value="Submit" />
```

Comment:
This works great, however when I type gibberish, it keeps the repos from the previous successful search, without throwing an error.